index.js

```diff
@@ -103,40 +110,58 @@ module.exports = function(options){
-                // only the first file should have @imports
-                if(!options.imports || index){
-                  return contents;
-                }
-
-                var imports = '';
+            var addImports = function(aFile, fileNamesOfPartsToImport){
                 var parameters = options.cacheBuster ? '?z=' + Math.round((Math.random() * 999)) : '';
-                for (var i = 1; i < numberOfSplits; i++) {
-                    imports += "@import url('" + createBlessedFileName(i) + parameters + "');\n\n";
+                var filePath = path.relative(aFile.base, aFile.path);
+                var concat = new Concat(shouldCreateSourcemaps, filePath, '\n');
+                for (var i = 0; i < fileNamesOfPartsToImport.length; i++) {
+                    concat.add(null, "@import url('" + fileNamesOfPartsToImport[i] + parameters + "');\n");
                 }
 
-                return imports + contents;
-            };
+                concat.add(filePath, aFile.contents, JSON.stringify(aFile.sourceMap));
 
+                aFile.contents = concat.content;
+                if (shouldCreateSourcemaps) {
+                    aFile.sourceMap = JSON.parse(concat.sourceMap);
+                }
+            };
 
             var outputFiles = [];
-            for(var j = numberOfSplits - 1; j >= 0; j--) {
-                var newIndex = numberOfSplits - 1 - j;
-                var outputPath = newIndex
-                    ? path.resolve(path.join(outputPathStart, createBlessedFileName(newIndex)))
-                    : outputFilePath;
+            var nonMasterPartFileNames = [];
+            for(var j = 0; j < numberOfSplits; j++) {
+                var oneBasedIndex = j + 1;
+                var isAtLastElement = j == numberOfSplits - 1;
+
+                //last element is the "master" file (the one with @import).
+                var outputPath = isAtLastElement
+                    ? outputFilePath
+                    : path.resolve(path.join(outputPathStart, createBlessedFileName(oneBasedIndex)));
 
-                outputFiles[newIndex] = addSourcemap(new File({
+                var outFile = addSourcemap(new File({
                     cwd: file.cwd,
                     base: file.base,
                     path: outputPath,
-                    contents: new Buffer(addImports(newIndex, result.data[j]))
+                    contents: new Buffer(result.data[j])
                 }), j);
-            }
+        
+                if (options.imports) {
+                    if (isAtLastElement) {
+                        addImports(outFile, nonMasterPartFileNames);
+                    } else {
+                        nonMasterPartFileNames.push(path.basename(outputPath)); 
+                    }
+                }
 
+                outputFiles[j] = outFile;
+            }
 
+            //We want to stream the "master" file first then split part1, part2, part3 ... this is mainly to maintain backward
```

Comment:
Actually it's the end of the file that has the `@imports`. This is so that the original unsplit CSS selectors order can be preserved if we put `@imports` at the top. For example lets say we have following CSS:

```
//original.css
.style1 {}
.style2 {}
.style3{}
```

which gets split into
```
//original-blessed1.css
.style1 {}
``` 

```
//original-blessed3.css
.style2 {}
```

```
//origin.css
@import(original-blessed1.css)
@import(original-blessed2.css)

.style3{}
```